lib/katello/engine.rb

```diff
@@ -195,6 +195,15 @@ def find_assets(args = {})
       ::Api::V2::HostsController.send :include, Katello::Concerns::Api::V2::HostsControllerExtensions
       ::Api::V2::HostgroupsController.send :include, Katello::Concerns::Api::V2::HostgroupsControllerExtensions
 
+      #facet extensions
+      Facets.register(Katello::Host::ContentFacet, :content_facet) do
+        api_view :list => 'katello/api/v2/content_facet/base_with_root', :single => 'katello/api/v2/content_facet/show'
+      end
+
+      Facets.register(Katello::Host::SubscriptionFacet, :subscription_facet) do
+        api_view :list => 'katello/api/v2/subscription_facet/base_with_root', :single => 'katello/api/v2/subscription_facet/show'
+      end
```

Comment:
well the problem is that the facets are going to be used within foreman core.  Now you could have a facet registration api that maybe could be used inside and out of the plugin api but that just adds complexity.